### PR TITLE
nghttp2: 1.40.0 + bump depepdencies + fix configure

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **nghttp2/1.40.0**

- Depends on #475 : jemalloc is enabled by default
- Switched to cmake unconditionally

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries 
I think this is a conan bug in the `pkg_config` generator:
The `libxml2.pc` file has the following contents on my system:
```
acec3/libxml-2.0.pc 
prefix=/home/maarten/.conan/data/libxml2/2.9.9/_/_/package/d482af2e6c91256ee2ba320636e8092512a15779
libdir=${prefix}/lib
includedir=${prefix}/include/libxml2

Name: libxml2
Description: libxml2 is a software library for parsing XML documents
Version: 2.9.9
Libs: -L${libdir} -lxml2  -lm  -Wl,-rpath="${libdir}"
Cflags: -I${includedir} -DLIBXML_STATIC
Requires: ZLIB libiconv
```

Notice the uppercase `ZLIB` in the `Requires` section.
It should be lower cased.

**Comment by danimtb** [comment]
@madebr should names in the `Requires:` section of pc files be always in lowercase? Maybe it is just a matter of adding `self.cpp_info.names["pkg_config"] = "zlib"` to the zlib recipe

**Comment by madebr** [comment]
@danimtb The `pkg_config` generator created a `zlib.pc` file so the item in the require section should also be `zlib`.

**Comment by danimtb** [comment]
@madebr I am having a look and this looks like a bug as you pointed out. I will open an issue with this case. Thanks a lot!

File: recipes/libnghttp2/all/conanfile.py [hunk 1]
```diff
@@ -10,45 +10,52 @@ class Nghttp2Conan(ConanFile):
...
+            del self.options.fPIC
+        if self.options.with_asio and self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Build with asio and MSVC is not supported yet, see upstream bug #589")
         if self.settings.compiler == "gcc":
             v = tools.Version(str(self.settings.compiler.version))
             if v < "6.0":
                 raise ConanInvalidConfiguration("gcc >= 6.0 required")
 
-    def config_options(self):
-        if self.settings.os == 'Windows':
```
Comment (uilianries) [inline recipes/libnghttp2/all/conanfile.py]: revert it, fPIC must be removed on Windows

File: recipes/libnghttp2/all/patches/fix-findLibevent.cmake [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- cmake/FindLibevent.cmake
```
Comment (uilianries) [inline recipes/libnghttp2/all/patches/fix-findLibevent.cmake]: Why not use `cmake_find_package` generator instead?

**Comment by danimtb** [comment]
Please solve the conflicts and update this PR to the latest changes in master. Thanks!

**Pull request merged**